external/storm-jdbc/src/test/java/org/apache/storm/jdbc/topology/AbstractUserTopology.java

```diff
@@ -0,0 +1,102 @@
+    private static final List<String> setupSqls = Lists.newArrayList(
+            "create table if not exists user (user_id integer, user_name varchar(100), dept_name varchar(100), create_date date)",
+            "create table if not exists department (dept_id integer, dept_name varchar(100))",
+            "create table if not exists user_department (user_id integer, dept_id integer)",
+            "insert into department values (1, 'R&D')",
+            "insert into department values (2, 'Finance')",
+            "insert into department values (3, 'HR')",
+            "insert into department values (4, 'Sales')",
+            "insert into user_department values (1, 1)",
+            "insert into user_department values (2, 2)",
+            "insert into user_department values (3, 3)",
+            "insert into user_department values (4, 4)"
+    );
+    protected UserSpout userSpout;
+    protected JdbcMapper jdbcMapper;
+    protected JdbcLookupMapper jdbcLookupMapper;
+
+    protected static final String TABLE_NAME = "user";
+    protected static final String JDBC_CONF = "jdbc.conf";
+    protected static final String SELECT_QUERY = "select dept_name from department, user_department where department.dept_id = user_department.dept_id" +
+            " and user_department.user_id = ?";
+
+    public void execute(String[] args) throws Exception {
+        if (args.length != 4 && args.length != 5) {
+            System.out.println("Usage: " + this.getClass().getSimpleName() + " <dataSourceClassName> <dataSource.url> "
+                    + "<user> <password> [topology name]");
+            System.exit(-1);
+        }
+        Map map = Maps.newHashMap();
+        map.put("dataSourceClassName", args[0]);//com.mysql.jdbc.jdbc2.optional.MysqlDataSource
+        map.put("dataSource.url", args[1]);//jdbc:mysql://localhost/test
+        map.put("dataSource.user", args[2]);//root
+        map.put("dataSource.password", args[3]);//password
+
+        Config config = new Config();
+        config.put(JDBC_CONF, map);
+
+        JDBCClient jdbcClient = new JDBCClient(map);
+        for (String sql : setupSqls) {
+            jdbcClient.executeSql(sql);
+        }
+
+        this.userSpout = new UserSpout();
+        this.jdbcMapper = new SimpleJdbcMapper(TABLE_NAME, map);
+        Fields outputFields = new Fields("user_id", "user_name", "dept_name", "create_date");
+        List<Column> queryParamColumns = Lists.newArrayList(new Column("user_id", Types.INTEGER));
+        this.jdbcLookupMapper = new SimpleJdbcLookupMapper(outputFields, queryParamColumns);
+
+        if (args.length == 4) {
+            LocalCluster cluster = new LocalCluster();
+            cluster.submitTopology("test", config, getTopology());
+            Thread.sleep(30000);
+            cluster.killTopology("test");
+            cluster.shutdown();
+            System.exit(0);
+        } else {
+            StormSubmitter.submitTopology(args[5], config, getTopology());
```

Comment:
yes, fixed. Sorry about this, originally I required that the user passed the table name as argument but I figured it is better to just execute all the DDLs as part of test so I removed it.